packages/components/src/color-palette/index.tsx

```diff
@@ -234,18 +237,26 @@ function UnforwardedColorPalette(
 		: __( 'Custom color picker.' );
 
 	const paletteCommonProps = {
-		clearable,
 		clearColor,
 		onChange,
 		value,
-		actions: !! clearable && (
-			<CircularOptionPicker.ButtonAction onClick={ clearColor }>
-				{ __( 'Clear' ) }
-			</CircularOptionPicker.ButtonAction>
-		),
-		headingLevel,
 	};
 
+	const actions = !! clearable && (
+		<CircularOptionPicker.ButtonAction onClick={ clearColor }>
+			{ __( 'Clear' ) }
+		</CircularOptionPicker.ButtonAction>
+	);
+
+	let ariaProps: { 'aria-label': string } | { 'aria-labelledby': string };
+	if ( ariaLabel ) {
+		ariaProps = { 'aria-label': ariaLabel as string };
+	} else if ( ariaLabelledby ) {
+		ariaProps = { 'aria-labelledby': ariaLabelledby as string };
+	} else {
+		ariaProps = { 'aria-label': __( 'Custom color picker.' ) };
+	}
```

Comment:
The `as string` type casts shouldn't be necessary:

```suggestion
	let ariaProps: { 'aria-label': string } | { 'aria-labelledby': string };
	if ( ariaLabel ) {
		ariaProps = { 'aria-label': ariaLabel };
	} else if ( ariaLabelledby ) {
		ariaProps = { 'aria-labelledby': ariaLabelledby };
	} else {
		ariaProps = { 'aria-label': __( 'Custom color picker.' ) };
	}
```

Reply:
I wonder if I changed something somewhere, because I'm sure at one point it was complaining that the value was `string or undefined` and so couldn't be coerced into a `string`. Ho hum. Will fix.